### ISSUE TITLE
Feature Request: Support for h2o models from R

### ISSUE THREAD

**Issue by djacobs7** [mlflow/mlflow#1423]
### Description
I am using the h2o library to create models, and I would like to save the model to mlflow.  There seems to be support for this in python ([here](https://mlflow.org/docs/latest/models.html#h2o-h2o
)), but not yet in R.

### Source Code
Presumably a code snippet would look like this:
 ```
   library(h2o)
    h2o.init()
    as.h2o( iris, destination_frame = 'iris')
    randomForest = h2o.randomForest( x = 'Sepal.Length', y ='Sepal.Width', training_frame = 'iris')
    mlflow_save_model(randomForest )

```
### System information

* R package mlflow_1.0.0 
* h2o.getVersion() = "3.24.0.1"
 

@kevinykuo

**Comment by augustomarcolin** [comment]
are there any updates about this issue? @smurching

**Comment by lorenzwalthert** [comment]
I implemented support for xgboost in #3085 (following the keras model implementation), so I think it wouldn't be too hard to adapt for h2o.

**Comment by smurching** [comment]
Yeah, we'd be happy to review a contribution adding support for logging h2o models from R. Following the approach used by the [existing keras flavor](https://github.com/mlflow/mlflow/blob/master/mlflow/R/mlflow/R/model-keras.R) seems like a reasonable way to go, would be happy to discuss further if anybody's interested in contributing :)

**Comment by yl790** [comment]
Hey @smurching I'm interested in taking on this task as an exercise to get up-to-speed with mlflow. I'll let you know if I run into any question.

**Comment by lorenzwalthert** [comment]
Great @yl790. I just pushed some tests I forgot to push to my [xgboost PR](https://github.com/mlflow/mlflow/pull/3085/files). I think these tests could be taken more or less 1 to 1 for h2o - and they are more comprehensive than the ones for keras, namely because they also test non-R / R API interactions.

**Comment by lorenzwalthert** [comment]
xgboost PR merged, let's proceed with more model flavours 🎉 .

**Comment by yitao-li** [comment]
@smurching @lorenzwalthert I created [PR](https://github.com/mlflow/mlflow/pull/3416) for this and saw it working in my local dev env, but there are a lot of unrelated failures from other tests in Github CI (mostly along the lines of http 404/experiment does not exist, etc). I wonder if I'm missing something.

**Comment by lorenzwalthert** [comment]
This should be considered closed with #3416 (should probably have used `closes #[PR number]` syntax to automatically close it with the PR).  @yitao-li I think it would be cool to implement more backends, i.e. sparklyr/MLeap integration or lightgbm, following the same pattern as #3416. Also note that there is a tidymodels flavor discussion in #3418.

**Comment by yitao-li** [comment]
@dmatrix I think we can close this issue now :+1:
